server/gfmd/db_access.c

```diff
@@ -765,6 +765,117 @@ db_user_load(void *closure, void (*callback)(void *, struct gfarm_user_info *))
 	return (e);
 }
 
+void *
+db_user_auth_dup(const struct db_user_auth_arg *ua, size_t size)
+{
+	struct db_user_auth_arg *r;
+	size_t usize = strlen(ua->username) + 1;
+	size_t msize = strlen(ua->auth_method) + 1;
+	size_t asize = strlen(ua->auth_user_id) + 1;
+	size_t sz;
+	int overflow = 0;
+
+#ifdef __GNUC__ /* workaround gcc warning: might be used uninitialized */
+	r = NULL;
+#endif
+	sz = gfarm_size_add(&overflow, size, usize + msize + asize);
+	if (!overflow)
+		r = malloc(sz);
+	if (overflow || r == NULL) {
+		gflog_debug(GFARM_MSG_UNFIXED,
+			"allocation of 'db_user_auth_arg' failed or overflow");
+		return (NULL);
+	}
+	r->username = (char *)r + size;
+	r->auth_method = r->username + usize;
+	r->auth_user_id = r->auth_method + msize;
+
+	strcpy(r->username, ua->username);
+	strcpy(r->auth_method, ua->auth_method);
+	strcpy(r->auth_user_id, ua->auth_user_id);
+	return (r);
+}
+
+void *
+db_user_auth_remove_dup(const struct db_user_auth_remove_arg *ua, size_t size)
+{
+	struct db_user_auth_remove_arg *r;
+	size_t usize = strlen(ua->username) + 1;
+	size_t msize = strlen(ua->auth_method) + 1;
+	size_t sz;
+	int overflow = 0;
+
+#ifdef __GNUC__ /* workaround gcc warning: might be used uninitialized */
+	r = NULL;
+#endif
+	sz = gfarm_size_add(&overflow, size, usize + msize);
+	if (!overflow)
+		r = malloc(sz);
+	if (overflow || r == NULL) {
+		gflog_debug(GFARM_MSG_UNFIXED,
+			"allocation of 'db_user_auth_remove_arg' failed or overflow");
```

Comment:
Please fold this line,
e.g.
```
	    "allocation of 'db_user_auth_remove_arg'"
 	    "failed or overflow");
```

because "WARNING: line over 80 characters" is reported by checkpatch.pl.

Note that second level indents are 4 spaces instead of a TAB
as written in doc/internal/ja/gfarm-coding-style.txt (from NetBSD KNF rev 1.20)